pinecone/index.py

```diff
@@ -108,61 +108,50 @@ def upsert(
         namespace: Optional[str] = None,
         batch_size: Optional[int] = None,
         show_progress: bool = True,
-        **kwargs
+        **kwargs,
     ) -> UpsertResponse:
         """
         The upsert operation writes vectors into a namespace.
         If a new value is upserted for an existing vector id, it will overwrite the previous value.
 
-        API reference: https://docs.pinecone.io/reference/upsert
-
         To upsert in parallel follow: https://docs.pinecone.io/docs/insert-data#sending-upserts-in-parallel
 
+        A vector can be represented by a 1) Vector object, a 2) tuple or 3) a dictionary
```

Comment:
There was lots of formatting jank in here, did a brief pass for now.

Reply:
We can handle it in a follow-up PR because I'm sure there are a lot of other places where this comes up, but in general we want to migrate most "vector" references to "record" to align with the [docs here](https://docs.pinecone.io/docs/overview#pinecone-indexes-store-records-with-vector-data).